jwst/dq_init/dq_init_step.py

```diff
@@ -14,48 +14,52 @@ class DQInitStep(Step):
 
     reference_file_types = ['mask']
 
-    def process(self, input):
-
-        with datamodels.open(input) as input_model:
-
-            # Check for consistency between keyword values and data shape
-            nints, ngroups, ysize, xsize = input_model.data.shape
-            nints_kwd = input_model.meta.exposure.nints
-            ngroups_kwd = input_model.meta.exposure.ngroups
-            ysize_kwd = input_model.meta.subarray.ysize
-            xsize_kwd = input_model.meta.subarray.xsize
-            if nints != nints_kwd:
-                self.log.error("Keyword 'NINTS' value of '{0} does not match data array size of '{1}'".format(nints_kwd,nints))
-                raise ValueError("Bad data dimensions")
-            if ngroups != ngroups_kwd:
-                self.log.error("Keyword 'NGROUPS' value of '{0}' does not match data array size of '{1}'".format(ngroups_kwd,ngroups))
-                raise ValueError("Bad data dimensions")
-            if ysize != ysize_kwd and 'IRS2' not in input_model.meta.exposure.readpatt.upper():
-                self.log.error("Keyword 'SUBSIZE2' value of '{0}' does not match data array size of '{1}'".format(ysize_kwd,ysize))
-                raise ValueError("Bad data dimensions")
-            if xsize != xsize_kwd:
-                self.log.error("Keyword 'SUBSIZE1' value of '{0}' does not match data array size of '{1}'".format(xsize_kwd,xsize))
-                raise ValueError("Bad data dimensions")
-
-            # Retreive the mask reference file name
-            self.mask_filename = self.get_reference_file(input_model, 'mask')
-            self.log.info('Using MASK reference file %s', self.mask_filename)
-
-            # Check for a valid reference file
-            if self.mask_filename == 'N/A':
-                self.log.warning('No MASK reference file found')
-                self.log.warning('DQ initialization step will be skipped')
-                result = input_model.copy()
-                result.meta.cal_step.dq_init = 'SKIPPED'
-                return result
-
-            # Load the reference file
-            mask_model = datamodels.MaskModel(self.mask_filename)
-
-            # Apply the step
-            result = dq_initialization.correct_model(input_model, mask_model)
-
-            # Close the reference file
-            mask_model.close()
+    def process(self, input): 
+
+        input_model = datamodels.open( input )
+        exp_type_pref = input_model.meta.exposure.type[:3]
+        if exp_type_pref == 'FGS':  # reopen as a GuiderRawModel 
```

Comment:
This logic will need to be expanded, because the normal (science) modes of FGS also start with the 3 letters "FGS", such as FGS_IMAGE, FGS_DARK, and FGS_FOCUS. The only ones that are exclusive to guiding operations are FGS_ID-IMAGE, FGS_ID-STACK, FGS_ACQ1, FGS_ACQ2, FGS_TRACK, and FGS_FINEGUIDE. Those values could be put into a list and then just test to see if exp_type_pref has a value in that list.